### ISSUE TITLE
A list variable have inconsistant ordering

### ISSUE THREAD

**Issue by yamamoto-febc** [hashicorp/terraform#10747]
### Terraform Version
v0.8.0

### Affected Resource(s)
Core

### Terraform Configuration Files
```hcl
resource "docker_image" "image" {
  name = "alpine"
}

resource docker_container "arg10" {
  name = "arg10"
  image = "${docker_image.image.latest}" 
  entrypoint = ["/usr/bin/yes"]
  command = ["0","1","2","3","4","5","6","7","8","9"]
}

resource docker_container "arg11" {
  name = "arg11"
  image = "${docker_image.image.latest}" 
  entrypoint = ["/usr/bin/yes"]
  command = ["0","1","2","3","4","5","6","7","8","9","10"]
}

output arg10{
  value = ["${docker_container.arg10.command}"]
}
output arg11{
  value = ["${docker_container.arg11.command}"]
}
```

### Debug Output
```sh
arg10 = [
    0,
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9
]
arg11 = [
    0,
    1,
    10,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9
]
```

### Expected Behavior
List order should be consistent and independent on number of items in list

### Actual Behavior
When 11 or more items are included in the list, order was disturbed

### Steps to Reproduce
1. `terraform apply`

**Comment by martin-flaregames** [comment]
Looks like as if the output got sorted using an alphanumeric collecting order

**Comment by jbardin** [comment]
The outputs here are sorted in lexicographical order, based on the address key of the value internally. 

The order is only shown incorrectly in the output, but I'll see if we can clean that up for display.

**Comment by jbardin** [comment]
Actually I was mistaken, it's not only in this order for display. 

The interpolated list is stored in the incorrect order, which means that the indexing order will be incorrect for interpolated lists, both within a module and through outputs.

**Comment by jbardin** [comment]
closed via #10765
